flake.nix

```diff
@@ -3,23 +3,39 @@
 
   description = "Home Manager for Nix";
 
-  outputs = { self, nixpkgs }: rec {
-
-    nixosModules.home-manager = import ./nixos nixpkgs;
-
-    lib = {
-      homeManagerConfiguration = { configuration, system, homeDirectory
-        , username
-        , pkgs ? builtins.getAttr system nixpkgs.outputs.legacyPackages
-        , check ? true }@args:
-        import ./modules nixpkgs {
-          pkgs = builtins.getAttr system nixpkgs.outputs.legacyPackages;
-          configuration = { ... }: {
-            imports = [ configuration ];
-            home = { inherit homeDirectory username; };
+  outputs = { self, nixpkgs }:
+    let
+      systems = nixpkgs.lib.platforms.unix;
+
+      forAllSystems = f: nixpkgs.lib.genAttrs systems (system: f system);
+
+      # Memoize nixpkgs for different platforms for efficiency.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        });
+    in rec {
```

Comment:
I belive `rec` should be removed here, and self should be used instead where it is needed

Reply:
I don't think there even is any self-reference anywhere. Can probably be removed

Reply:
removed it with the current rebase